### DISCUSSION TITLE
Preview with DRM on top layer

### DISCUSSION THREAD

**Discussion by brochn** [raspberrypi/picamera2#686] [Q&A]
Hi, 

I am trying to migrate Pibooth app on Raspberry Pi Lite from Picamera to Picamera2. The software uses PyGame as graphical library and I am working without X. In Picamera, there was the property that starting the preview with start_preview() was on top-layer of the image and was therefore overwriting some part of the PyGame image (on purpose). With Picamera2, I haven't found how to reproduce that and therefore, the consequence is that when I use start_preview(), the preview image is (I guess), hidden by PyGame. Does anyone has an idea in which direction I should investigate to be able to include preview image in PyGame?
Thanks for your help.
Br, Nicolas

**Comment by davidplowman** [comment]
Hi, when you say that you're working "without X", you mean that you're running applications from the console, with no GUI at all? I wasn't aware that you can use PyGame without X Windows, but then I don't know anything about it. I'm aware that I can get camera windows to appear on top of everything else using the DRM preview, but couldn't say how that interacts with anything else. If there's no one else who reads this discussion list who can help, then you might have more luck posting to one of the more general Raspberry Pi forums? Sorry not to be more help.

**Reply by brochn** [reply]
Hi, thanks for your answer. 
indeed I'm working directly with framebuffer. So there is no GUI at all. Then, I will dive a bit more in the topic and look whether I can get work with the help of Rpi forum.
Have a nice day.
